src/rust/engine/fs/fs_util/src/main.rs

```diff
@@ -187,10 +203,22 @@ fn execute(top_match: clap::ArgMatches) -> Result<(), ExitError> {
           Ok(materialize_directory(store, destination, &fingerprint)?)
         }
         ("save", Some(args)) => {
-          let posix_fs = Arc::new(make_posix_fs(args.value_of("source").unwrap()));
-          let (fingerprint, size_bytes) =
-            save_directory(store, posix_fs, Arc::new(fs::Dir(PathBuf::from("."))))
-              .wait()?;
+          let (root, source) = extract_root_and_source(args).map_err(|e| {
+            ExitError(
+              format!("Error parsing root and source flags: {}", e.description()),
+              ExitCode::UnknownError,
+            )
+          })?;
+          let posix_fs = Arc::new(make_posix_fs(root));
+          let (fingerprint, size_bytes) = posix_fs
+            .expand(fs::PathGlobs::create(
+              &[format!("{}/**", source.to_str().unwrap())],
```

Comment:
I'm not sure whether I should be accepting a directory, or a glob here. I vaguely feel like this should accept a directory (which I implicitly upgrade to a glob, as I do here), and there is scope for a separate command along the lines of

`fs_util save <globs>`

which accepts any number of glob patterns.

WDYT?